.github/workflows/ansible-test.yml

```
@@ -58,8 +58,40 @@ jobs:
         with:
           fail_ci_if_error: false
 
+  integration_matrix:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ansible_collections/community/mongodb
+    outputs:
+      matrix: ${{ steps.json-list.outputs.modules-to-test }}
+
+    steps:
+
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          path: ansible_collections/community/mongodb
+
+      - name: Generate module list
+        run: |
+          find tests/integration/targets -type d -maxdepth 1 -not -name 'setup*' | cut -d '/' -f4 | sort | awk NF > int_test_modules.txt
+      - name: Remove last newline from file
+        run: |
+          perl -pi -e 'chomp if eof' int_test_modules.txt
+      - name: Generate json doc from module list
+        run: |
+          cat int_test_modules.txt | jq -sR 'split("\n")' > int_test_modules.json
```

```suggestion
          cat int_test_modules.txt | jq -sRc 'split("\n")' > int_test_modules.json
```
Using `jq -c` will do compact output (print the array on one line).
I think that's the point of the python in the next step, right?
If you use `-c` maybe you can just cat the file in the set-output command?

---

That should work. Then I can remove the perl task.